dashboard/app/controllers/registrations_controller.rb

```diff
@@ -163,13 +163,14 @@ def create
           error_message: "retry ##{retries} failed with exception: #{exception}"
         )
       end
-      super
-    end
 
-    if params[:new_sign_up].present?
-      session[:user_return_to] ||= params[:user_return_to]
-      @user = Services::PartialRegistration::UserBuilder.call(request: request)
-      sign_in @user
+      if params[:new_sign_up].present? && params[:new_sign_up] == true
```

Comment:
I figured out what was happening here- it appears the param type is a string, so it was recognizing 'false' as present and throwing all the tests into the new registration create flow. By forcing it to check for true explicitly, I was able to add puts statements and verify that only the tests I expected were making it through

Reply:
Oh wait but if you're seeing the tests you'd expect with the parameter you'd expect then I'm wrong! Maybe could we do that Boolean assigning just to be safe? I'm just worried with this boolean/string type checking with what will pass as `true` because it's a non-empty string or `true` because it's actual value is `true`

Reply:
I should've mentioned that I added a commit this morning that was checking for the string! But yes, happy to cast to a boolean instead. I double checked this worked and went back to just using present? on the casted boolean.

Reply:
Sounds good! I believe the `.present?` isn't needed anymore either since casting `nil` to boolean still returns `nil` so it won't enter that `if` statement

Reply:
Gahh you're right. Removed

Reply:
Thank you!

Reply:
Thank you!!!